src/progress/web/method/cgidefs.i

```diff
@@ -0,0 +1,175 @@
+DEFINE NEW GLOBAL SHARED VARIABLE HostURL           AS char FORMAT "x(40)":U 
+    NO-UNDO.
+
+/* Default path for set-cookie() function.  Normally the same as AppURL.
+   Otherwise, obtained from the DefaultCookiePath configuration option. */
+DEFINE NEW GLOBAL SHARED VARIABLE CookiePath        AS char FORMAT "x(40)":U 
+    NO-UNDO.
+
+/* Relative path for our static HTML, JAVA, DOC, and IMG file off the web
+   server's DOCUMENT_ROOT directory.                                    */   
+DEFINE NEW GLOBAL SHARED VARIABLE RootURL           AS char FORMAT "x(60)":U 
+    NO-UNDO.
+
+/* Messenger Server Connection ID is inactive/active (0/1)              */   
+DEFINE NEW GLOBAL SHARED VARIABLE useConnID         AS char FORMAT "x(60)":U 
+    NO-UNDO.
+
+/* Default domain for the set-cookie() function.  Normally blank. */
+DEFINE NEW GLOBAL SHARED VARIABLE CookieDomain      AS char FORMAT "x(40)":U 
+    NO-UNDO.
+
+/* Delimiter for multiple items assigned to a single field name.
+   Typically a comma or tab. */
+DEFINE NEW GLOBAL SHARED VARIABLE SelDelim          AS char FORMAT "x":U 
+    NO-UNDO INITIAL ",":U.
+
+/* Flag set by the output-content-type() function to the Content-Type header
+   value that was output.  If this is blank, then the output-content-type()
+   function has not been run yet which should imply no other output has
+   been sent either. */
+DEFINE NEW GLOBAL SHARED VARIABLE output-content-type AS char NO-UNDO.
+
+/* Newline characters to use for the output-http-header() function.
+   Some web servers do not allow the CR, LF combination but only support
+   an LF character. */
+DEFINE NEW GLOBAL SHARED VARIABLE http-newline AS char NO-UNDO
+    INITIAL "~r~n":U.
+
+/* Offset from Coordinated Universal Time (UTC) or GMT time in seconds.
+   This value is updated with each request.  A positive number is west of
+   UTC.  This is used to calculate UTC time when setting Cookie expiration
+   dates in the set-cookie() function. */
+DEFINE NEW GLOBAL SHARED VARIABLE utc-offset AS INTEGER NO-UNDO INITIAL -1.
+
+/* Handle for web utilities procedure. */
+DEFINE NEW GLOBAL SHARED VARIABLE web-utilities-hdl AS HANDLE NO-UNDO.
+
+/* Comma delimited list of debugging options. */
+DEFINE NEW GLOBAL SHARED VARIABLE debug-options AS char FORMAT "x(20)":U
+    NO-UNDO.
+
+/* True if debugging is enabled via the configuration options Environment
+   and Debugging. */
+DEFINE NEW GLOBAL SHARED VARIABLE debugging-enabled AS LOGICAL NO-UNDO
+    INITIAL TRUE.
+
+/* List of Misc. variables above.  Used for dumping out all misc. values in
+   debugging mode such as with printval.p. */
+DEFINE NEW GLOBAL SHARED VARIABLE MiscVarList     AS char NO-UNDO 
+    INITIAL "AppProgram,AppURL,HostURL,SelDelim,SelfURL,":U.
+
+&GLOBAL-DEFINE OUT-STREAM PUT {&WEBSTREAM} UNFORMATTED
+/* OUT-FILE is the same as OUT-STREAM, for now.  */
+&GLOBAL-DEFINE OUT-FILE {&OUT-STREAM}
+&GLOBAL-DEFINE OUT {&OUT-STREAM}
+&GLOBAL-DEFINE OUT-LONG EXPORT {&WEBSTREAM}
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) From src/progress/pct/pctWSComp.p - Preprocessor variable out-long should be SCOPED-DEFINED [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)